### ISSUE TITLE
lineno needs to be loaded after amsmath

### ISSUE THREAD

**Issue by FrankMittelbach** [latex-lineno/lineno#5]
With the new version lineno supports the  amsmath environments (thanks for that!) but unfortunately it introduces an order dependency.
```
\documentclass{article}
\usepackage{lineno}
\usepackage{amsmath}

\begin{document}
aaa
\end{document}

errors with 
! LaTeX Error: Command \gather already defined.
```
It might be better (and in fact much simpler) to make use LaTeX's  hook system and make the alterations automatically after amsmath got loaded, via
```
\AddToHook{package/amsmath/after}{....}
```
In this case the changes happen only after the environments got defined in amsmath and not at all if amsmath is not loaded ever.

**Comment by daamien** [comment]
related: https://tex.stackexchange.com/questions/669008/package-minted-and-tcolorbox-produces-latex-error-command-gather-already-defin

**Comment by jfine2358** [comment]
Related: "conflict between lineno and amsmath packages in TeXLive 2022? Some environments (like \gather) are already defined", title of post https://tug.org/pipermail/texhax/2023-January/025962.html.

In that thread on texhax I suggest (see https://tug.org/pipermail/texhax/2023-January/025967.html)

> How about asking TeXLive to revert lineno to the previous version (which it seems does not have this conflict with amsmath). Perhaps overall this would make the best of what there is.

**Comment by kwwette** [comment]
Fixed in version 5.1, which should appear on CTAN soon.
- For LaTeX releases after October 2020, `lineno` now patches `amsmath` with `\AddToHook`, so that the packages can be loaded in any order.
- For LaTeX releases before October 2020 (where `\AddToHook` is unavailable), `lineno` will only try to patch `amsmath` if it is already loaded, otherwise it will print a warning. This will still allow the packages to be loaded in any order, expect that `amsmath` will not be patched if it is loaded after `lineno`, and hence line numbers won't work correctly in `amsmath` math environments.
